spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -0,0 +1,361 @@
+require 'spec_helper'
+require 'datadog/core/crashtracking/component'
+
+require 'webrick'
+require 'fiddle'
+
+RSpec.describe Datadog::Core::Crashtracking::Component,
+  skip: !CrashtrackingHelpers.supported? do
+    describe '.build' do
+      let(:settings) { Datadog::Core::Configuration::Settings.new }
+      let(:agent_settings) { double('agent_settings') }
+      let(:logger) { Logger.new($stdout) }
+      let(:tags) { { 'tag1' => 'value1' } }
+      let(:agent_base_url) { 'agent_base_url' }
+      let(:ld_library_path) { 'ld_library_path' }
+      let(:path_to_crashtracking_receiver_binary) { 'path_to_crashtracking_receiver_binary' }
+
+      context 'when all required parameters are provided' do
+        it 'creates a new instance of Component and starts it' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+          expect(logger).to_not receive(:warn)
+
+          component = double(instance_double(described_class))
```

Comment:
I'm not even quite sure what `double(instance_double(...))` does, but I suspect it may not be the right thing ;)

```suggestion
          component = instance_double(described_class)
```